builder/azure/arm/builder.go

```diff
@@ -82,11 +83,21 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	resolver := newResourceResolver(azureClient)
+	if err := resolver.Resolve(b.config); err != nil {
+		return nil, err
+	}
+
 	b.config.storageAccountBlobEndpoint, err = b.getBlobEndpoint(azureClient, b.config.ResourceGroupName, b.config.StorageAccount)
 	if err != nil {
 		return nil, err
 	}
 
+	endpointConnectType := PublicEndpoint
```

Comment:
You don't necessarily need a public IP as you can still use the private IP if you are connected to the pre-existing environment via VPN and you can still use your provisioner. This is a needed use case for our current setup which requires this feature to make this possible.